handlers/product-move-api/src/main/scala/com/gu/productmove/endpoint/available/AvailableProductMovesEndpoint.scala

```diff
@@ -44,44 +53,149 @@ object AvailableProductMovesEndpoint {
       .serverLogic[TIO] { subscriptionName => run(subscriptionName).tapEither(result => ZIO.log("result tapped: " + result)).map(Right.apply) }
   }
 
+  // sub to test on: "A-S00334930"
   private def run(subscriptionName: String): TIO[OutputBody] =
-    runWithEnvironment(subscriptionName).provide(
+    runWithEnvironment("A-S00334930").provide(
       AwsS3Live.layer,
       AwsCredentialsLive.layer,
       SttpClientLive.layer,
       ZuoraClientLive.layer,
       GetSubscriptionLive.layer,
+      GetCatalogueLive.layer,
+      GetAccountLive.layer,
       ZuoraGetLive.layer,
       GuStageLive.layer,
     )
 
-  private[productmove] def runWithEnvironment(subscriptionName: String): ZIO[GetSubscription, String, OutputBody] =
+  private val freeTrialDays = 14
+
+
+  def getAvailableSwitchRatePlans(zuoraProductCatalogue: ZuoraProductCatalogue, ratePlanNames: List[String]): List[ZuoraProductRatePlan] =
+    val productRatePlans = for {
+      product <- zuoraProductCatalogue.products.filter(ratePlanNames contains _.name)
+      productRatePlan <- product.productRatePlans.toList
+    } yield productRatePlan
+
+    productRatePlans.find(_.name == "Digital Pack Monthly").toList
+
+  val localDateToString = DateTimeFormatter.ISO_LOCAL_DATE
+
+  def succeedIfEligible(accountEligible: Boolean, message: String): ZIO[Any, Unit, Unit] =
+    if (accountEligible)
+      ZIO.succeed(())
+    else
+      for {
+        _ <- ZIO.log(message)
+        _ <- ZIO.fail(())
+      } yield ()
+
+  def getCreditCardExpirationDate(wireDefaultPaymentMethod: WireDefaultPaymentMethod): Either[Unit, LocalDate] =
     for {
+      creditCardExpirationMonth <- wireDefaultPaymentMethod.creditCardExpirationMonth.toRight(())
+      creditCardExpirationYear <- wireDefaultPaymentMethod.creditCardExpirationYear.toRight(())
+    } yield LocalDate.of(creditCardExpirationYear, creditCardExpirationMonth, 1)
+
+  extension[R, E, A] (zio: ZIO[R, E, A])
+    def handleError(message: String) = zio.catchAll {
```

Comment:
let's change this to failureAs500
```suggestion
    def failureAs500(message: String) = zio.catchAll {
```